objects/src/accounts/mod.rs

```diff
@@ -154,6 +140,19 @@ impl Account {
         self.nonce == ZERO
     }
 
+    /// Returns hash of the given account as used for the initial account state hash in transaction
+    /// proofs.
+    ///
+    /// For existing accounts, this is exactly the same as [Account::hash()], however, for new
+    /// accounts this value is set to `None`. This is because when a transaction is executed
+    /// against a new account, public input for the initial account state is set to [ZERO; 4] to
+    /// distinguish new accounts from existing accounts, and `None` returned from this method.
+    /// indicates that `[ZERO; 4]` should be used as a public input. The actual hash of the initial
+    /// account state (and the initial state itself), are provided to the VM via the advice provider.
+    pub fn init_account_hash(&self) -> Option<Digest> {
+        self.is_new().not().then(|| self.hash())
+    }
```

Comment:
I would move this method back where it was (i.e., line 93 above). Regarding naming, I'd probably just call it `init_hash()` (I think the `account` part is not needed as it is a method on `Account` struct anyway).